gfx_scene.h

```diff
@@ -1744,35 +1744,47 @@ class GfxSceneInternal
-            is_node_animated = (is_node_animated || is_any_children_animated);
+            {
+                auto const [is_child_animated, child_animation] = VisitNode(gltf_node.children[i], transform, is_node_animated, node_animation);
+                GFX_ASSERT(!is_any_children_animated || !is_child_animated || children_animation == child_animation); // we assume non overlapping animations
+                is_any_children_animated = is_any_children_animated || is_child_animated;
+                children_animation = is_child_animated ? child_animation : children_animation;
+            }
+            if (!is_node_animated && is_any_children_animated)
+            {
+                is_node_animated = true;
+                node_animation = children_animation;
+            }
             if(is_node_animated)
             {
                 GltfNode *node = nullptr;
                 std::vector<uint64_t> children;
                 GltfAnimatedNode *animated_node = nullptr;
                 for(size_t i = 0; i < gltf_node.children.size(); ++i)
                 {
-                    std::map<int32_t, uint64_t>::const_iterator const it2 = animated_nodes.find(gltf_node.children[i]);
-                    if(it2 != animated_nodes.end()) children.push_back((*it2).second);
+                    std::map<int32_t, std::pair<size_t, uint64_t>>::const_iterator const it2 = animated_nodes.find(gltf_node.children[i]);
+                    if(it2 != animated_nodes.end()) children.push_back((*it2).second.second);
                 }
                 if(it != animated_nodes.end())
                 {
-                    unparented_nodes.erase((*it).second);
-                    node = &gltf_nodes_[GetObjectIndex((*it).second)];
-                    animated_node = gltf_animated_nodes_.at(GetObjectIndex((*it).second));
-                    gltf_node_refs_.insert(GetObjectIndex((*it).second), (uint32_t)animations.size());
+                    unparented_nodes.erase((*it).second.second);
+                    node = &gltf_nodes_[GetObjectIndex((*it).second.second)];
+                    animated_node = gltf_animated_nodes_.at(GetObjectIndex((*it).second.second));
+                    gltf_node_refs_.insert(GetObjectIndex((*it).second.second), (uint32_t)animations.size());
                 }
                 else
                 {
+                    GFX_ASSERT(node_animation < animations.size());
                     uint64_t const animated_node_handle = gltf_node_handles_.allocate_handle();
                     gltf_node_refs_.insert(GetObjectIndex(animated_node_handle), (uint32_t)animations.size());
                     node = &gltf_nodes_.insert(GetObjectIndex(animated_node_handle));
-                    animated_nodes[node_index] = animated_node_handle;
+                    animated_nodes[node_index] = {node_animation, animated_node_handle};
```

Comment:
`node_animation` is an index, it can be the animation of the current node (if available) or the animation of one parent or child (if missing).